File: 1. Arrays & Hashing/3. Two Sum/dory.py
# Runtime 1ms
# Beats 98.81%
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {}
        for i in range(len(nums)):
            buffer = target - nums[i]
            if buffer in hashmap:
                return [hashmap[buffer], i]
            else:
                hashmap[nums[i]] = i
        return []

File: 1. Arrays & Hashing/3. Two Sum/dory.cs
// Runtime 1ms
// Beats 99.02%

// Memory 49.25MB
// Beats 23.45%

public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> dict = new Dictionary<int, int>();
        for(int i = 0; i < nums.Length; i++) {
            int c = target - nums[i];
            if(dict.ContainsKey(c)) {
                return new int[] {dict[c], i};
            } else {
                dict[nums[i]] = i;
            }
        }
        return new int[] { };
    }
}